### PR TITLE
Features/redux toolkit

### PR DESCRIPTION
**Pull request by shirrotem**

### CODE REVIEW

File: src/shared/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 }
 
-const Modal: React.FC<ModalProps> = ({ onClose, children }) => {
+const Modal: React.FC<ModalProps> = ({ children }) => {
+  const { isOpen }= useAppSelector(state=> state.homePage);
+  const dispatch= useAppDispatch();
+
+  const onClose = () => {
+    dispatch(setIsOpen(!isOpen));
+  };
```
Comment (MoranOshiaMoveo) [inline src/shared/components/Modal/Modal.tsx]: use useCallback

File: src/shared/components/Card/Card.tsx [hunk 2]
```diff
@@ -1,18 +1,35 @@
...
+    if(isDish){
+        handleOnClick = useCallback((dishName: string = '') => {
+        const dish = data.dishes.find(dishObj => dishObj.name === dishName);
+        if(dish){
+          dispatch(setSelectedDish(dish));
+        }
+        dispatch(setIsOpen(!isOpen));
+      }, [data.dishes, isOpen]);
+    }
+
```
Comment (MoranOshiaMoveo) [inline src/shared/components/Card/Card.tsx]: set it as func with useCallaback

File: src/shared/components/Card/Card.tsx [hunk 3]
```diff
@@ -1,18 +1,35 @@
...
     return (
       <div onClick={isDish && handleOnClick ? ()=> handleOnClick(title) : undefined} className={`card ${isRestaurant ? 'restaurant-card' : isDish ? 'dish-card' : isChef ? 'chef-card' : isDishModal ? 'card-modal-content' : ''}`}>
```
Comment (MoranOshiaMoveo) [inline src/shared/components/Card/Card.tsx]: the is dish is unnecessary u already checked it on the on click func, also the if is not relevant

File: src/pages/home-page/components/Carousel/Carousel.tsx [hunk 4]
```diff
@@ -7,10 +7,22 @@ import useMobileCheck from "../../../../shared/hooks/useMobileCheck";
...
+  let type: any;
+  if(typeName==="chefRestaurants"){
+    type= data.chefOfTheWeek.chefRestaurants;
+  }
+  else if(typeName==="dishes"){
+    type= data.dishes;
+  }
+  else if(typeName==="restaurants"){
+    type= data.restaurants;
+  }
```
Comment (MoranOshiaMoveo) [inline src/pages/home-page/components/Carousel/Carousel.tsx]: what is the propose of this terms

**Review by MoranOshiaMoveo — CHANGES_REQUESTED** [review]

File: src/shared/components/Modal/Modal.tsx (hunk 1 above)
Comment (shirrotem) [reply src/shared/components/Modal/Modal.tsx]: fixed

File: src/shared/components/Card/Card.tsx (hunk 3 above)
Comment (shirrotem) [reply src/shared/components/Card/Card.tsx]: fixed

File: src/shared/components/Card/Card.tsx (hunk 2 above)
Comment (shirrotem) [reply src/shared/components/Card/Card.tsx]: fixed

File: src/pages/home-page/components/Carousel/Carousel.tsx (hunk 4 above)
Comment (shirrotem) [reply src/pages/home-page/components/Carousel/Carousel.tsx]: fixed

File: src/pages/home-page/components/Carousel/Carousel.tsx [hunk 5]
```diff
@@ -7,10 +7,20 @@ import useMobileCheck from "../../../../shared/hooks/useMobileCheck";
 import arrowsIcon from "../../../../assets/icons/arrows.svg";
 import './Carousel.scss';
 import CarouselProps from "../../../../data/types/carouselProps";
+import { useAppSelector } from "../../../../redux-toolkit/store/store";
 
 
-const Carousel: FC<CarouselProps> = ({ type, typeName, title, handleOnClick}) => {
+const Carousel: FC<CarouselProps> = ({typeName, title}) => {
```
Comment (MoranOshiaMoveo) [inline src/pages/home-page/components/Carousel/Carousel.tsx]: change to obj instead of type name

**Review by MoranOshiaMoveo — CHANGES_REQUESTED** [review]

File: src/pages/home-page/components/Carousel/Carousel.tsx (hunk 5 above)
Comment (shirrotem) [reply src/pages/home-page/components/Carousel/Carousel.tsx]: fixed

**Review by MoranOshiaMoveo — APPROVED** [review]

**Pull request merged**